dna-src/zomes/communities/code/src/communities/mod.rs

```diff
@@ -19,22 +19,42 @@ pub struct Community {
     pub slug: String,
 }
 
+#[derive(Serialize, Deserialize, Debug, Clone, DefaultJson)]
+pub struct CommunityResult {
+    pub address: Address,
+    pub name: String,
+    pub slug: String
+}
+
 pub type Base = RawString;
 
 const COMMUNITY_BASE_ENTRY: &str = "community_base";
-const COMMUNITY_LINK_TAG: &str = "member_of";
+const COMMUNITY_LINK_TYPE: &str = "member_of";
+
+pub fn get(address: Address) -> ZomeApiResult<CommunityResult> {
+    let community: Result<Community, _> = utils::get_as_type(address.clone());
 
-pub fn get_community(address: Address) -> ZomeApiResult<Community> {
-    utils::get_as_type(address)
+    match community {
```

Comment:
seems likely there's a neater way to do this pattern than an explicit match here, especially with a function to convert a Community into CommunityResult

Reply:
```rust
utils::get_as_type::<Community, _>(address.clone())
   .map(|community| {
       community.with_address(address)
   })
```
will give a `ZomeApiResult<CommunityWithAddress>` which makes the function one expression. It also has the bonus of preserving the errors messages from core

Reply:
great!